### ISSUE TITLE
Need to reboot twice for Hass.io to start

### ISSUE THREAD

**Issue by 1iveowl** [home-assistant/hassio#915]
**Home Assistant release with the issue:**
0.86.4 - but have seen the issue consistently with all releases the past months. 
Hass.io supervisor version: 143, but seen this with previous versions also the past months

**Operating environment (HassOS/Generic):**
Ubuntu Server LTS 18.4.1

**Description of problem:**
I’ve seen this behavior consistently over the past months.

Every time to reboot my Ubuntu Server 18.04.1, the first time it comes back online Hass is unable to start.

Specifically, the homeassistant/amd64-hassio-supervisor keep recreating itself every approx. 6 sec. This can be spotted using ‘docker ps -a’.

I then reboot Ubuntu Server again and Hass.io starts the second time. Also, the first time I reboot it usually takes Ubuntu Server longer time to come back online and accessable using SSH.

What is causing this and how do I fix it?

**Comment by pvizeli** [comment]
what do you see on `journalctl -fu hassio-supervisor`?

**Comment by 1iveowl** [comment]
This is what I see when running `journalctl -fu hassio-supervisor`

> -- Logs begin at Sun 2018-12-02 14:22:44 CET. --
> Feb 08 10:51:22 ubuntusrv1 systemd[1]: Stopped Hass.io supervisor.
> Feb 08 10:51:22 ubuntusrv1 systemd[1]: Starting Hass.io supervisor...
> Feb 08 10:51:22 ubuntusrv1 docker[3564]: hassio_supervisor
> Feb 08 10:51:22 ubuntusrv1 systemd[1]: Started Hass.io supervisor.
> Feb 08 10:51:22 ubuntusrv1 hassio-supervisor[3586]: time="2019-02-08T10:51:22+01:00" level=error msg="error waiting for container: context canceled"
> Feb 08 10:51:22 ubuntusrv1 hassio-supervisor[3586]: Error response from daemon: error while creating mount source path '/usr/share/hassio': mkdir /usr/share/hassio: read-only file system
> Feb 08 10:51:22 ubuntusrv1 hassio-supervisor[3586]: hassio_supervisor
> Feb 08 10:51:23 ubuntusrv1 hassio-supervisor[3586]: docker: Error response from daemon: error while creating mount source path '/usr/share/hassio': mkdir /usr/share/hassio: read-only file system.
> Feb 08 10:51:23 ubuntusrv1 systemd[1]: hassio-supervisor.service: Main process exited, code=exited, status=125/n/a
> Feb 08 10:51:23 ubuntusrv1 systemd[1]: hassio-supervisor.service: Failed with result 'exit-code'.
> Feb 08 10:51:28 ubuntusrv1 systemd[1]: hassio-supervisor.service: Service hold-off time over, scheduling restart.
> Feb 08 10:51:28 ubuntusrv1 systemd[1]: hassio-supervisor.service: Scheduled restart job, restart counter is at 2.
> Feb 08 10:51:28 ubuntusrv1 systemd[1]: Stopped Hass.io supervisor.
> Feb 08 10:51:28 ubuntusrv1 systemd[1]: Starting Hass.io supervisor...
> Feb 08 10:51:28 ubuntusrv1 docker[3724]: hassio_supervisor
> Feb 08 10:51:28 ubuntusrv1 systemd[1]: Started Hass.io supervisor.
> Feb 08 10:51:28 ubuntusrv1 hassio-supervisor[3747]: time="2019-02-08T10:51:28+01:00" level=error msg="error waiting for container: context canceled"


Above is one instance. It just repeats endlessly from there.

After I reboot it works. After the reboot the log shows below.

Also, at the second reboot the server is much quicker to boot.

> -- Logs begin at Sun 2018-12-02 14:22:44 CET. --
> Feb 08 10:53:41 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:41 INFO (MainThread) [hassio.auth] Home Assistant not running, check cache
> Feb 08 10:53:41 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:41 INFO (MainThread) [hassio.auth] Cache hit for mqttuser
> Feb 08 10:53:41 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:41 INFO (SyncWorker_18) [hassio.docker.addon] Start Docker add-on homeassistant/amd64-addon-ssh with version 5.1
> Feb 08 10:53:42 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:42 INFO (SyncWorker_7) [hassio.docker.addon] Start Docker add-on homeassistant/amd64-addon-duckdns with version 1.5
> Feb 08 10:53:42 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:42 INFO (MainThread) [hassio.auth] Auth request from core_mosquitto for mqttuser
> Feb 08 10:53:42 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:42 ERROR (MainThread) [hassio.homeassistant] Error on call https://172.30.32.1:8123/api/: Cannot connect to host 172.30.32.1:8123 ssl:None [Connection refused]
> Feb 08 10:53:42 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:42 INFO (MainThread) [hassio.auth] Home Assistant not running, check cache
> Feb 08 10:53:42 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:42 INFO (MainThread) [hassio.auth] Cache hit for mqttuser
> Feb 08 10:53:47 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:47 INFO (SyncWorker_9) [hassio.docker.interface] Clean homeassistant/qemux86-64-homeassistant Docker application
> Feb 08 10:53:47 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:47 INFO (SyncWorker_9) [hassio.docker.homeassistant] Start homeassistant homeassistant/qemux86-64-homeassistant with version 0.87.0
> Feb 08 10:54:07 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:07 INFO (MainThread) [hassio.homeassistant] Detect a running Home Assistant instance
> Feb 08 10:54:07 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:07 INFO (MainThread) [hassio.addons] Startup application run 2 add-ons
> Feb 08 10:54:07 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:07 INFO (SyncWorker_19) [hassio.docker.interface] Clean homeassistant/amd64-addon-configurator Docker application
> Feb 08 10:54:07 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:07 INFO (SyncWorker_7) [hassio.docker.interface] Clean hassioaddons/node-red-amd64 Docker application
> Feb 08 10:54:08 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:08 INFO (SyncWorker_7) [hassio.docker.addon] Start Docker add-on hassioaddons/node-red-amd64 with version v1.3.0
> Feb 08 10:54:08 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:08 INFO (SyncWorker_19) [hassio.docker.addon] Start Docker add-on homeassistant/amd64-addon-configurator with version 2
> Feb 08 10:54:09 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:54:09 INFO (MainThread) [hassio.api.security] /addons access from a0d7b954_nodered

**Comment by 1iveowl** [comment]
Does this provide any insights into what may be the issue?

**Comment by unlucio** [comment]
I'm a noob to hassio but this looks like a clue for why it didn't start the 1st time:
```
Feb 08 10:51:23 ubuntusrv1 hassio-supervisor[3586]: docker: Error response from daemon: error while creating mount source path '/usr/share/hassio': mkdir /usr/share/hassio: read-only file system.
```
I'd check the boot sequence to see why `/usr/share/` seems been read-only at that point in time.

I then found this (your second problem and looks like the one I'm experiencing):
```
Feb 08 10:53:42 ubuntusrv1 hassio-supervisor[1763]: 19-02-08 09:53:42 ERROR (MainThread) [hassio.homeassistant] Error on call https://172.30.32.1:8123/api/: Cannot connect to host
```
which means it cannot connect/find to any homeassistant instance (port 8123), and I'd love to know why the supervisor doesn't start it.
I'm having the same result in a slightly different setup where `hassio_supervisor` doesn't start nor recreate the homeassistant's container.
@pvizeli isn't `hassio_supervisor` supposed to "supervise" homeassistant? :D

**Comment by unlucio** [comment]
One more thing: I installed it using this script: `https://github.com/home-assistant/hassio-build/tree/master/install`
which pulled the following images:
`homeassistant/armhf-hassio-supervisor:143`
`homeassistant/armhf-hassio-supervisor:latest`
`homeassistant/raspberrypi3-homeassistant:0.87.1`
`homeassistant/raspberrypi3-homeassistant:latest`

**Comment by 1iveowl** [comment]
Thank you @unlucio 

This this count be the root cause:

> Feb 08 10:51:23 ubuntusrv1 hassio-supervisor[3586]: docker: Error response from daemon: error while creating mount source path '/usr/share/hassio': mkdir /usr/share/hassio: read-only file system.

But I have not idea what is causing this and how to fix it.

**Comment by 1iveowl** [comment]
I found the issue. 

Mea culpa. I had two versions of Docker installed:

- The official Docker version installed with apt-get
- and a version installed with snap, which I think was installed when I first installed Ubuntu.

Anyhow, just removing the snap version of Docker did the trick.

Here's how:

1. Run `sudo snap list` to see verify that Docker is installed with snap. If you see it on the list, it is.
2. Then run: `snap remove docker` 
3. Reboot.
